src/brpc/checksum.h

```diff
@@ -0,0 +1,62 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#ifndef BRPC_CHECKSUM_H
+#define BRPC_CHECKSUM_H
+
+#include <google/protobuf/message.h>  // Message
+
+#include "brpc/controller.h"
+#include "brpc/options.pb.h"  // ChecksumType
+#include "butil/iobuf.h"      // butil::IOBuf
+
+namespace brpc {
+
+struct ChecksumHandler {
+    // checksum `buf'.
+    // Returns checksum value
+    void (*Compute)(const butil::IOBuf& buf, Controller* cntl);
+
+    // verify buf checksum
+    // Rerturn true on success, false otherwise
+    bool (*Verify)(const butil::IOBuf& buf, const Controller& cntl);
```

Comment:
Do we need to design parameters to be more extensible? Like #2946, the parameters may be expanded in the future.

Reply:
For example, use struct instead of all parameters:

```c++
struct In {
    const butil::IOBuf& buf; 
    Controller* cntl
};
void (*Compute)(const In& in);
bool (*Verify)(const In& in);
```